### PR TITLE
Remove rescheduler and corresponding tests from master

### PR DESCRIPTION
**Pull request by ravisantoshgudimetla**
**What this PR does / why we need it**:
This is to remove rescheduler from master branch as we are promoting priority and preemption to beta.
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of #57471

**Special notes for your reviewer**:
/cc @bsalamat @aveshagarwal 
**Release note**:

```release-note
Remove rescheduler from master.
```

### CODE REVIEW

File: test/e2e/scheduling/equivalence_cache_predicates.go [hunk 1]
```diff
@@ -35,6 +35,10 @@ import (
 	_ "github.com/stretchr/testify/assert"
 )
 
+const (
+	defaultTimeout = 3 * time.Minute
```
Comment (bsalamat) [inline test/e2e/scheduling/equivalence_cache_predicates.go]: Why is this needed? It does not seem to belong to this PR.

**Review by bsalamat — COMMENTED** [review]
Looks good overall. Please make sure that the test failures are not caused by this PR.

File: test/e2e/scheduling/equivalence_cache_predicates.go (hunk 1 above)
Comment (ravisantoshgudimetla) [reply test/e2e/scheduling/equivalence_cache_predicates.go]: The `defaultTimeout` from rescheduler.go(which I removed) is being used in equivalence_cache_predicates.go :(.  So, had to add it here..

>  Please make sure that the test failures are not caused by this PR.

Thanks for the review. Looks like they are not related. I will rebase the PR later in the evening and will ask the bot to test the changes again.

**Comment by ravisantoshgudimetla** [comment]
@bsalamat The PR seems to not have problem with any of the tests. Can you add v1.11 milestone.

**Review by bsalamat — COMMENTED** [review]
/lgtm

**Review by ixdy — APPROVED** [review]
/approve

**Comment by derekwaynecarr** [comment]
reviewed the kubelet changes.

/lgtm
/approve

**Comment by ravisantoshgudimetla** [comment]
Thanks @bsalamat @derekwaynecarr @eparis @ixdy
